miniapp/miniapp.cpp

```diff
@@ -67,7 +68,9 @@ int main(int argc, char** argv) {
             global_policy::set_sizes(options.dry_run_ranks, cells_per_rank);
         }
 
-        banner();
+        hw::node nd;
+        nd.num_cpu_cores = threading::num_threads();
```

Comment:
I agree more or less with this. Here is what I have done
* `node_info` had its smarts removed: it is just a container.
* added a `get_node_info()` function that will return a `node_info` with the number of cores and gpus using the `hw` namespace
* in the miniapp I set `nd.num_cpu_cores` to the number of threads used by the threading back end: `threading::num_threads()`. Because the threading back end currently chooses the number of threads using its own internal logic from parsing environment variables and the `hw::` interface, I leave it like this... in future this will change when users are responsible for initializing the threading backend.
* If there is more than one gpu available, use one gpu.